### PR TITLE
Improve Digital adapter: Add deal id and rewardred inventory logic

### PR DESCRIPTION
**Pull request by jbartek25**
- Handle Rewarded inventory flag
- Add dealid

### CODE REVIEW

File: adapters/improvedigital/improvedigital.go [hunk 1]
```diff
@@ -13,10 +14,27 @@ import (
...
+)
+
 type ImprovedigitalAdapter struct {
 	endpoint string
 }
 
+// BidExt This struct usage for parse line_item_id and buying_type from bid.ext
+type BidExt struct {
+	Improvedigital struct {
+		LineItemId int    `json:"line_item_id"`
```
Comment (AlexBVolcy) [inline adapters/improvedigital/improvedigital.go:33]: Nitpick: could you re-name this variable to be `LineItemID`

File: adapters/improvedigital/improvedigital.go [hunk 2]
```diff
@@ -222,3 +256,37 @@ func (a *ImprovedigitalAdapter) getAdditionalConsentProvidersUserExt(request ope
...
+	}
+
+	var prebidMap = make(map[string]json.RawMessage)
+	if err := json.Unmarshal(prebidJSONValue, &prebidMap); err != nil {
+		return nil, err
+	}
+
+	rewardedInventory, foundRewardedInventory := prebidMap[isRewardedInventory]
+	if !foundRewardedInventory {
+		return nil, nil
```
Comment (AlexBVolcy) [inline adapters/improvedigital/improvedigital.go:278]: I see this line isn't covered by tests. Is it possible to test this case?

File: adapters/improvedigital/improvedigital.go [hunk 3]
```diff
@@ -36,6 +54,15 @@ func (a *ImprovedigitalAdapter) MakeRequests(request *openrtb2.BidRequest, reqIn
 }
 
 func (a *ImprovedigitalAdapter) makeRequest(request openrtb2.BidRequest, imp openrtb2.Imp) (*adapters.RequestData, error) {
+	// Handle Rewarded Inventory
+	extImp, err := getImpExtWithRewardedInventory(imp)
```
Comment (AlexBVolcy) [inline adapters/improvedigital/improvedigital.go:58]: Nitpick: Could you re-name this to be `impExt`

File: adapters/improvedigital/improvedigital.go (hunk 2 above)
Comment (samiul-shanto) [reply adapters/improvedigital/improvedigital.go:278]: Test case added

**Review by mansinahar — DISMISSED** [review]

**Comment by jbartek25** [comment]
Sorry for the mess. I thought this PR was already merged. Reopening.

**Review by mansinahar — COMMENTED** [review]
Other than the minor comment above, LGTM

**Review by mansinahar — APPROVED** [review]

**Review by AlexBVolcy — APPROVED** [review]

**Pull request merged**